torchx/lib/torchx/backend.ex

```diff
@@ -756,6 +756,14 @@ defmodule Torchx.Backend do
     |> to_nx(out)
   end
 
+  @impl true
+  def clip(%T{} = out, %T{} = t, %T{} = min, %T{} = max) do
+    t
+    |> from_nx()
+    |> Torchx.clip(to_number(min), to_number(max))
+    |> to_nx(out)
```

Comment:
```suggestion
    t
    |> from_nx()
    |> Torchx.clip(to_number(min), to_number(max))
    |> to_nx(out)
```
```suggestion
    t
    |> Nx.as_type(out.type)
    |> from_nx()
    |> Torchx.clip(from_nx(min), from_nx(max))
    |> to_nx(out)
```

I've applied my suggestions locally and they work fine, but this conversion is needed beforehand :)